Shared/Extensions/URLExtensions.swift

```diff
@@ -237,6 +237,39 @@ extension URL {
         return publicSuffixFromHost(host, withAdditionalParts: 1)
     }
 
+    /**
+    * Returns a possible mobile host and the domain, but with the same scheme, and a trailing '/'.
+    *
+    * E.g., https://m.foo.com/bar/baz?noo=abc#123  => https://m.foo.com/
+    * Or https://mobile.foo.com/bar/baz?noo=abc#123 => https://mobile.foo.com/
+    */
+    public var mobileDomainURL: URL {
```

Comment:
All of this code is duplicated and the only difference is between `"^(www)\\."`, `"^(www|mobile|m)\\."` regexes
What do you think about some DRY refactor?

- change `domainURL` to a function `func domainURL(stripWWWSubdomainOnly: Bool = false) -> URL`
- same for `normalizedHost`
- in `normalizedHost` add something like `let regex = stripWWWSubdomainOnly ? "^(www)\\." : "^(www|mobile|m)\\."`
- update documentation comment in `domainURL`
- update `testdomainURL` unit test

Reply:
Great suggestions! I've made the requested changes and added a new unit test to account for the `stripWWWSubdomainOnly` parameter.